### PR TITLE
Endpoint documentation for the Areas v2 service

### PR DESCRIPTION
**Pull request by ikas**
Update on the v2 area endpoints' documentation.

**Does not include work on the v1 area endpoints.**

~~Missing concept documentation :facepalm:~~ concept docs in a separate PR :arrow_right: https://github.com/resource-watch/doc-api/pull/159/files

Related to this JIRA issue: https://gfw.atlassian.net/browse/GTC-280

### CODE REVIEW

File: source/includes/api/_areas_v2.md [hunk 1]
```diff
@@ -1,8 +1,20 @@
...
+## What is an area?
 
-Please ensure that you are using `v2` in the URL when requesting these endpoints.
+Areas is a service provided by the RW API which allows you to define geographic areas of interest. Areas can be defined in multiple ways, such as selecting a whole country or region as your area of interest, a protected area from WDPA, or reference a custom geostore ID from the [Geostore service](#geostore).
+
+With your areas of interest, you can then choose to be notified about events that happen inside your area of interest, such as deforestation alerts (GLAD alerts) or fire alerts (VIIRS fire alerts). You can also choose to receive a monthly summary of both GLAD and VIIRS alerts in your area of interest, which will arrive in your inbox at the first day of every month.
+
+## What is the difference between v1 and v2?
+
+v2 areas are an upgrade in the functionality of areas and provide you with an easier-to-use interface for creating areas of interest. Features such as the notification of alerts inside you area of interest are also only available in v2 endpoints.
```
Comment (ethanroday) [inline source/includes/api/_areas_v2.md]:
```suggestion
v2 areas are an upgrade in the functionality of areas and provide you with an easier-to-use interface for creating areas of interest. Features such as the notification of alerts inside your area of interest are also only available in v2 endpoints.
```

File: source/includes/api/_areas_v2.md [hunk 2]
```diff
@@ -221,25 +289,24 @@ curl -X POST https://api.resourcewatch.org/v2/area
...
-webhookUrl           | Webhook URL to be provided to the subscription (only used in case the email is not set).           | String  | https://www.google.com/ |
-status               | The status of the area - either 'saved' or 'pending'. Read-only attribute.                         | String  | 'saved'    |
-subscriptionId       | The ID of the subscription associated with this area. Read-only attribute.                         | String  | 5e4d7c47dd8fa31290d548ae |
+Use this endpoint to create new areas. For a detailed description of each field that can be provided in the body of the request, check out the [Area reference](#area-reference) section.
+
+Keep in mind that you should provide one of the following when creating an area:
+
+* `geostore` with the ID of a geostore object if you are creating an area that references a geostore;
+* `wdpaid` with the ID of a protected area if you are creating an area that references a protected area;
+* `iso` object with a valid country/region/subregion if you are creating an area that references an admin country/region/subregion;
```
Comment (ethanroday) [inline source/includes/api/_areas_v2.md:313]: What is an `iso` object / `use` object? Are these not just strings? If not, do we have a reference for them anywhere?
Definitely could use more detail about what the `use` option is and where the possible values are defined.

File: source/includes/api/_areas_v2.md [hunk 3]
```diff
@@ -221,25 +289,24 @@ curl -X POST https://api.resourcewatch.org/v2/area
...
+
+### Errors for creating an area
+
+Error code     | Error message (example)     | Description
+-------------- | --------------------------- | ------------------------------------------------------------
+400            | `<field> can not be empty.` | You are missing a required field while creating the area.
+400            | `<field> is not valid.`     | You provided an invalid field while creating the area.
+401            | `Unauthorized`              | No token was provided.
 
 ### Email notification
```
Comment (ethanroday) [inline source/includes/api/_areas_v2.md]: I would move this "According to multiple factors..." paragraph to the end of the previous section

File: source/includes/api/_areas_v2.md [hunk 4]
```diff
@@ -257,17 +324,17 @@ The following parameters are provided to the email service and can be used in th
 * `id` : the ID of the area.
```
Comment (ethanroday) [inline source/includes/api/_areas_v2.md]: A little confused about this section. What does it mean that the parameters are "provided to the email service"? Do I specify them somewhere? If I'm a user creating an area, not sure how this is relevant to me.

File: source/includes/api/_areas_v2.md [hunk 5]
```diff
@@ -257,17 +324,17 @@ The following parameters are provided to the email service and can be used in th
...
+* `map_link` : the "view on map" URL for this area (example: [https://staging.globalforestwatch.org/map/aoi/5d517b3fb8cfd4001061d0b2](https://staging.globalforestwatch.org/map/aoi/5d517b3fb8cfd4001061d0b2)).
 * `image_url` : the URL for the image associated with the area.
 * `tags` : a string containing the AOI tags, comma-separated.
 
 (`5d517b3fb8cfd4001061d0b2` is an example of an area ID).
 
 ### Implementation details
 
-POST of a new area always starts by creating the area and, taking into account the area attributes, it might also create a subscription which will then be associated with the area. The area's `subscriptionId` attribute will then be updated with the id of the created subscription if that's the case. The created area is then returned.
+POST of a new area always starts by creating the area. Then, taking into account the area attributes, it might also create a subscription which will then be associated with the area. The area's `subscriptionId` attribute will then be updated with the id of the created subscription if that's the case. The created area is then returned.
```
Comment (ethanroday) [inline source/includes/api/_areas_v2.md:339]: Before, it said that area v2 was intended to replace area v1 + subscriptions. But it sounds like subscriptions are still created in area v2? This could use clarification.

File: source/includes/api/_areas_v2.md [hunk 6]
```diff
@@ -326,46 +393,25 @@ curl -X PATCH https://api.resourcewatch.org/v2/area/:id
...
-deforestationAlerts  | If the area is intended to subscribe to deforestation alerts. Defaults to false.                   | Boolean | true       |
-monthlySummary       | If the area is intended to subscribe to monthly summaries. Defaults to false.                      | Boolean | true       |
-email                | Email to be provided to the subscription.                                                          | String  | youremail@resourcewatch.org |
-webhookUrl           | Webhook URL to be provided to the subscription (only used in case the email is not set).           | String  | https://www.google.com/ |
-status               | The status of the area - either 'saved' or 'pending'. Read-only attribute.                         | String  | 'saved'    |
-subscriptionId       | The ID of the subscription associated with this area. Read-only attribute.                         | String  | 5e4d7c47dd8fa31290d548ae |
+Use this endpoint to update an existing area. For a detailed description of each field that can be provided in the body of the request, check out the [Area reference](#area-reference) section.
```
Comment (ethanroday) [inline source/includes/api/_areas_v2.md]: Nice job removing duplication 😄

File: source/includes/api/_areas_v2.md [hunk 7]
```diff
@@ -326,46 +393,25 @@ curl -X PATCH https://api.resourcewatch.org/v2/area/:id
...
 
-### Email notification
+Keep in mind that you should provide one of the following when updating an area:
```
Comment (ethanroday) [inline source/includes/api/_areas_v2.md]: Should or must?

File: source/includes/api/_areas_v2.md [hunk 8]
```diff
@@ -326,46 +393,25 @@ curl -X PATCH https://api.resourcewatch.org/v2/area/:id
...
-* `location` : an alias for the name of the area (contains the same as the `name` parameter).
-* `subscriptions_url` : the URL to manage the areas in the frontend (example: [https://staging.globalforestwatch.org/my-gfw](https://staging.globalforestwatch.org/my-gfw)).
-* `dashboard_link` : the link to the area dashboard (example: [https://staging.globalforestwatch.org/dashboards/aoi/5d517b3fb8cfd4001061d0b2](https://staging.globalforestwatch.org/dashboards/aoi/5d517b3fb8cfd4001061d0b2)).
-* `map_link` : the "view on map" for the area (example: [https://staging.globalforestwatch.org/map/aoi/5d517b3fb8cfd4001061d0b2](https://staging.globalforestwatch.org/map/aoi/5d517b3fb8cfd4001061d0b2)).
-* `image_url` : the URL for the image associated with the area.
-* `tags` : a string containing the AOI tags, comma-separated.
-
-(`5d517b3fb8cfd4001061d0b2` is an example of an area ID).
+Error code     | Error message (example)       | Description
+-------------- | ----------------------------- | --------------
```
Comment (ethanroday) [inline source/includes/api/_areas_v2.md:410]: Looks like this table needs updating. Messages are not parameterized and description still talks about creation.

**Review by ethanroday — CHANGES_REQUESTED** [review]
This looks good! See comments for some clarifications and revisions. Specifically, I think the relationship between areas and subscriptions could be explained a little better.

File: source/includes/api/_areas_v2.md (hunk 6 above)
Comment (ikas) [reply source/includes/api/_areas_v2.md]: Less is more :v:

File: source/includes/api/_areas_v2.md (hunk 7 above)
Comment (ikas) [reply source/includes/api/_areas_v2.md]: Must, otherwise it will return 401 Unauthorized - thanks for pointing out :+1:

File: source/includes/api/_areas_v2.md (hunk 8 above)
Comment (ikas) [reply source/includes/api/_areas_v2.md:410]: Done, thanks for pointing out :+1:

**Comment by ikas** [comment]
Hey Ethan, thanks a lot for the feedback you provided, extremely useful :)

I agree that the connection between areas and subscriptions is not clear enough at the moment, so I'll try to explain it a bit better! I'll also give some more detail into the different methods for creating and updating areas (WDPA, ISO, use, etc) - I'll let you know once this is ready for review again :+1:

File: source/includes/api/_areas_v2.md (hunk 5 above)
Comment (ikas) [reply source/includes/api/_areas_v2.md:339]: Yes - I think the idea I missed here is that Areas still make use of Subscriptions for actually sending notifications to users. Once you are aware of this, it's easier to understand that yes, subscriptions are still created in areas v2.

I've added a section to the introduction called "Interaction between Areas and Subscriptions" which should clarify this interaction.

File: source/includes/api/_areas_v2.md (hunk 2 above)
Comment (ikas) [reply source/includes/api/_areas_v2.md:313]: These fields are defined in the Area reference section, at the bottom. I've added here a link to the Area reference section, but over there we should have every info needed for interacting with these fields.

File: source/includes/api/_areas_v2.md (hunk 4 above)
Comment (ikas) [reply source/includes/api/_areas_v2.md]: These are the variables available for interpolation for someone building Sparkpost templates. This is just informative for someone interacting with the templates in Sparkpost, not useful for the "normal" API user. I've added some more information about this process above which hopefully should clarify this part.

**Comment by ikas** [comment]
@ethanroday all feedback has been addressed, the PR is once again up for review!

File: source/includes/api/_areas_v2.md [hunk 9]
```diff
@@ -1,8 +1,24 @@
 # Areas v2
 
-The following endpoints merge together **areas of interest** and **subscriptions**. This means they can be used to transition away from subscriptions without the users losing the subscriptions that they have already created.
+Before reading any further, please ensure you have read the [Areas of Interest concept documentation](#area) first. It gives you a brief and clear description of what an Area of Interest is and what it can do for you. 
 
-Please ensure that you are using `v2` in the URL when requesting these endpoints.
+Once you've read that section, you can come back here to learn more details about using the RW API's Areas service. Areas of interest are used by the Global Forest Watch website to subscribe to notifications on deforestation and fire alerts inside a particular areas you might be interest in. The sections below describe in detail how you can use the endpoints provided by RW API's [Areas service](https://github.com/gfw-api/gfw-area) to define your own geographic areas of interest.
```
Comment (tiagojsag) [inline source/includes/api/_areas_v2.md]: add a link to GFW

File: source/includes/api/_areas_v2.md [hunk 10]
```diff
@@ -45,6 +61,39 @@ curl -X GET https://api.resourcewatch.org/v2/area
 }
 ```
 
+The `v2/areas` endpoint returns all the areas of interest associated with the user who made the request. 
+
+**This endpoint requires authentication, and it will return 401 Unauthorized if none is provided.**
```
Comment (tiagojsag) [inline source/includes/api/_areas_v2.md]: Replace this type of notice with an "Errors for ..." section, like we've added to recent documentation refactors

Comment (tiagojsag) [reply source/includes/api/_areas_v2.md]: Also add other potential errors users may run into when using each respective endpoint

File: source/includes/api/_areas_v2.md [hunk 11]
```diff
@@ -221,53 +293,70 @@ curl -X POST https://api.resourcewatch.org/v2/area
...
+
+### Email notifications
+
+After creating an area, if the `email` field of the area contains a valid email, an email is sent to the user. The email content varies according to the status of the area:
 
 * If the area has status `saved`, an email is sent to let the user know the area of interest is ready to be viewed.
 * If the area has status `pending`, an email is sent to let the user know the area of interest is being generated and will be available later.
 
-### Email substitution parameters
+Emails are sent using [Sparkpost](https://www.sparkpost.com/) - so, for the emails to be sent, there should exist templates in Sparkpost ready to be sent, taking into account the different languages supported by the Areas service:
```
Comment (tiagojsag) [inline source/includes/api/_areas_v2.md]: Is this relevant to an API user? do they have access to sparkpost, or need to worry about it in any way?

File: source/includes/api/_areas_v2.md (hunk 4 above)
Comment (tiagojsag) [reply source/includes/api/_areas_v2.md]: Seems like this is api dev documentation, not api user. if that's the case, let's keep it simple here, and maybe add a section for api devs on this

File: source/includes/api/_areas_v2.md [hunk 12]
```diff
@@ -221,53 +293,70 @@ curl -X POST https://api.resourcewatch.org/v2/area
...
 
 * `id` : the ID of the area.
 * `name` : the name of the area.
 * `location` : an alias for the name of the area (contains the same as the `name` parameter).
-* `subscriptions_url` : the URL to manage the areas in the frontend (example: [https://staging.globalforestwatch.org/my-gfw](https://staging.globalforestwatch.org/my-gfw)).
-* `dashboard_link` : the link to the area dashboard (example: [https://staging.globalforestwatch.org/dashboards/aoi/5d517b3fb8cfd4001061d0b2](https://staging.globalforestwatch.org/dashboards/aoi/5d517b3fb8cfd4001061d0b2)).
-* `map_link` : the "view on map" for the area (example: [https://staging.globalforestwatch.org/map/aoi/5d517b3fb8cfd4001061d0b2](https://staging.globalforestwatch.org/map/aoi/5d517b3fb8cfd4001061d0b2)).
+* `subscriptions_url` : the URL for managing areas of interest in the flagship application (example: [https://staging.globalforestwatch.org/my-gfw](https://staging.globalforestwatch.org/my-gfw)).
```
Comment (tiagojsag) [inline source/includes/api/_areas_v2.md]: not sure we want to advertise the existence of a staging gfw site.

File: source/includes/api/_areas_v2.md [hunk 13]
```diff
@@ -221,53 +293,70 @@ curl -X POST https://api.resourcewatch.org/v2/area
...
+* `dashboard_link` : the URL for the area dashboard (example: [https://staging.globalforestwatch.org/dashboards/aoi/5d517b3fb8cfd4001061d0b2](https://staging.globalforestwatch.org/dashboards/aoi/5d517b3fb8cfd4001061d0b2)).
+* `map_link` : the "view on map" URL for this area (example: [https://staging.globalforestwatch.org/map/aoi/5d517b3fb8cfd4001061d0b2](https://staging.globalforestwatch.org/map/aoi/5d517b3fb8cfd4001061d0b2)).
 * `image_url` : the URL for the image associated with the area.
 * `tags` : a string containing the AOI tags, comma-separated.
 
 (`5d517b3fb8cfd4001061d0b2` is an example of an area ID).
```
Comment (tiagojsag) [inline source/includes/api/_areas_v2.md]: replace this with the typical `<area id>``placeholder we're using in other examples.

**Review by tiagojsag — CHANGES_REQUESTED** [review]
Some improvements here and there, but overall I think we're in the right path.

another thing I noticed is the use of "implementation details" here and there, to describe actual functionality. IMO an implementation detail should not describe behavior, but decisions made on how to implement said behavior. "if you delete an area, a subscription is also deleted" IMO is not an implementation detail, it's core behavior of an endpoint.

File: source/includes/api/_areas_v2.md (hunk 10 above)
Comment (ikas) [reply source/includes/api/_areas_v2.md]: Done and done :+1:

File: source/includes/api/_areas_v2.md (hunk 12 above)
Comment (ikas) [reply source/includes/api/_areas_v2.md]: Updated the example links to point to GFW production URL :+1:

File: source/includes/api/_areas_v2.md (hunk 4 above)
Comment (ikas) [reply source/includes/api/_areas_v2.md]: Accepted Tiago's advice and moved the bits related to email notification parameters and all the Sparkpost information to the developer docs :)

File: source/includes/api/_areas_v2.md (hunk 11 above)
Comment (ikas) [reply source/includes/api/_areas_v2.md]: This is relevant to API developers, so moved to the developer docs. Also added some information on getting access to the Sparkpost account.

**Comment by ikas** [comment]
Moved the bits related to Sparkpost and email notification parameters to the developer docs, since it's only relevant for API developers, not API users :)

**Review by tiagojsag — APPROVED** [review]

**Review by ethanroday — APPROVED** [review]
Thanks for adding more details on the interaction between areas and subscriptions - very helpful.
